app/routes/product-details.$productSlug/route.tsx

```diff
@@ -22,6 +22,16 @@ export const loader = ({ params, request }: LoaderFunctionArgs) => {
     return getProductDetailsRouteData(params.productSlug, request.url);
 };
 
+export const getStaticRoutes = async () => {
+    const products = await getEcomApi().getAllProducts();
+
+    if (products.status === 'failure') {
+        throw new Error(`${products.error.code} ${products.error.message || ''}`);
```

Comment:
I'll make sure to refactor this so message isn't optional.

If the message isn't optional, do you still need the code?
* If yes, let's have something like `formatEcomApiError(products.error)`
* If not, let's keep it as is, and I'll change to `new Error(products.error.message)` or ideally `throw prodcuts.error`;

Reply:
The `error.message` property is always defined now, see #165.

Concatenation of error code and message will results in things like "ProductNotFound Product not found".

This should be fine now: `throw new Error(products.error.message)`